make_ocr_features/make_OCR_df_and_preprocess.ipynb

```diff
@@ -0,0 +1,1070 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### \"Intersection words trick\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from utils.OCR_word_selection import get_intersect_words_lang, get_intersect_words_ocr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ratio: 0.14667169558293522\n"
+     ]
+    }
+   ],
+   "source": [
+    "\"\"\"takes approx 1min30\"\"\"\n",
+    "intersection_list_lang = parallel_calc(get_intersect_words_lang, lang_dict_list)\n",
+    "print(\"ratio:\", len([item for item in intersection_list_lang if item != \"\"]) / len(intersection_list_lang))\n",
+    "\n",
+    "df[\"intersection_words_lang\"] = intersection_list_lang"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ratio: 0.4644656776464672\n"
+     ]
+    }
+   ],
+   "source": [
+    "\"\"\"takes approx 56sec\"\"\"\n",
+    "sub_ocr_text_dict_list = [ocr_text_dict[code] for code in df[\"code\"]]\n",
+    "intersection_list_ocr = parallel_calc(get_intersect_words_ocr, sub_ocr_text_dict_list)\n",
+    "print(\"ratio:\", len([item for item in intersection_list_ocr if item != \"\"]) / len(intersection_list_ocr))\n",
+    "\n",
+    "df[\"intersection_words_ocr\"] = intersection_list_ocr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "164374    30\n",
+       "67550     30\n",
+       "279678    30\n",
+       "586477    30\n",
+       "244112    30\n",
+       "          ..\n",
+       "261679     0\n",
+       "261676     0\n",
+       "261675     0\n",
+       "261672     0\n",
+       "596549     0\n",
+       "Name: intersection_words_ocr, Length: 596550, dtype: int64"
+      ]
+     },
+     "execution_count": 26,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df[\"intersection_words_ocr\"].str.split().str.len().sort_values(ascending = False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fetch big words from images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gzip\n",
+    "import json\n",
+    "from utils.OCR_word_selection import get_big_words_from_txt_annotations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from collections import defaultdict\n",
+    "big_words_dict = defaultdict(str)\n",
```

Comment:
I'd move those vars in the next cell so they get properly reinitialized if next cell get re-run.